Changes for Cloud UI

dborkan opened the pull request:
Changes for Cloud UI.  There are a lot of TODOs in here, but I think we are in a state we this could be released as a MVP.  For all the "Learn more"/etc TODOs I'm waiting to get FAQ entries before adding these.  The other CSS TODOs I will work on this week but I don't think they should block merging this code.

Also this code doesn't have switch for the text shown on the initial splash screen or empty roster screen.  I can add that if we think it's really needed but I feel it probably isn't necessary (I can add it if we need to release a non-cloud uProxy in the next few weeks).  The digital ocean deployment screens still only show if ```model.cloudProviderNames.length > 0```

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/uproxy/uproxy/2313)
<!-- Reviewable:end -->

trevj commented:
Awesome. I love how this is looking and, more importantly, behaving.

FYI, I'm mostly ignored style changes - assuming you've discussed this with UX. We just received updated mocks from UX so worth double-checking the strings.

---

Review status: 0 of 35 files reviewed at latest revision, 19 unresolved discussions.

---

*[src/generic_ui/locales/en/messages.json, line 1167 \[r1\]](https://reviewable.io:443/reviews/uproxy/uproxy/2313#-KDU2QnQRlX-tvoQZCCz:-KDU2QnQRlX-tvoQZCD-:-853215326) ([raw file](https://github.com/uproxy/uproxy/blob/014cf92c9c9fec420808e0f5aa975ff1720dba60/src/generic_ui/locales/en/messages.json#L1167)):*
Worth including a comment as to what a cloud server is, for the benefit of translators?

---

*[src/generic_ui/locales/en/messages.json, line 1171 \[r1\]](https://reviewable.io:443/reviews/uproxy/uproxy/2313#-KDTuFEti6QPXWke_Ed7:-KDTuFEuoD5LAxD1orJt:28086915) ([raw file](https://github.com/uproxy/uproxy/blob/014cf92c9c9fec420808e0f5aa975ff1720dba60/src/generic_ui/locales/en/messages.json#L1171)):*
I believe it's DigitalOcean rather than Digital Ocean, throughout.

---

*[src/generic_ui/locales/en/messages.json, line 1175 \[r1\]](https://reviewable.io:443/reviews/uproxy/uproxy/2313#-KDTyYjpRLvochjAWTDE:-KDTyYjqiDtK9z4Ola02:-1692513969) ([raw file](https://github.com/uproxy/uproxy/blob/014cf92c9c9fec420808e0f5aa975ff1720dba60/src/generic_ui/locales/en/messages.json#L1175)):*
"Button text for signing into DigitalOcean"?

---

*[src/generic_ui/locales/en/messages.json, line 1182 \[r1\]](https://reviewable.io:443/reviews/uproxy/uproxy/2313#-KDTvCh9CFkGDQLAqj2p:-KDTvCh9CFkGDQLAqj2q:-200319716) ([raw file](https://github.com/uproxy/uproxy/blob/014cf92c9c9fec420808e0f5aa975ff1720dba60/src/generic_ui/locales/en/messages.json#L1182)):*
Many of the names, e.g. `DIGITAL_OCEAN_SUCCESS_TITLE`, reference DigitalOcean but could be re-used for any cloud provider. A more general name, e.g. `CLOUD_INSTALL_SUCCESS_TITLE`, would work just as well and be less surprising if/when we support additional providers.

---

*[src/generic_ui/locales/en/messages.json, line 1187 \[r1\]](https://reviewable.io:443/reviews/uproxy/uproxy/2313#-KDTvMZjjU2PiXc6n8GR:-KDTvMZjjU2PiXc6n8GS:2139459467) ([raw file](https://github.com/uproxy/uproxy/blob/014cf92c9c9fec420808e0f5aa975ff1720dba60/src/generic_ui/locales/en/messages.json#L1187)):*
This seems like the wrong description :-)

---

*[src/generic_ui/locales/en/messages.json, line 1188 \[r1\]](https://reviewable.io:443/reviews/uproxy/uproxy/2313#-KDTynB9lJu3IiuLVm4l:-KDTynBAhAIdLr_MAl03:-254139511) ([raw file](https://github.com/uproxy/uproxy/blob/014cf92c9c9fec420808e0f5aa975ff1720dba60/src/generic_ui/locales/en/messages.json#L1188)):*
Two spaces after a full stop?

---

*[src/generic_ui/locales/en/messages.json, line 1188 \[r1\]](https://reviewable.io:443/reviews/uproxy/uproxy/2313#-KDU3WQ_56MsQmSVNMoc:-KDU38RQ-oUP5sL_5IdR:200793669) ([raw file](https://github.com/uproxy/uproxy/blob/014cf92c9c9fec420808e0f5aa975ff1720dba60/src/generic_ui/locales/en/messages.json#L1188)):*
More of a UX/product question but this sentence is just weird. "cloud connection" -> "cloud server", for one thing.

---

*[src/generic_ui/locales/en/messages.json, line 1191 \[r1\]](https://reviewable.io:443/reviews/uproxy/uproxy/2313#-KDTyqsnnyvbgdgOx6sX:-KDTyqsoHtsnArLfgwaI:-1429804372) ([raw file](https://github.com/uproxy/uproxy/blob/014cf92c9c9fec420808e0f5aa975ff1720dba60/src/generic_ui/locales/en/messages.json#L1191)):*
I think this description can be more helpful.

---

*[src/generic_ui/locales/en/messages.json, line 1192 \[r1\]](https://reviewable.io:443/reviews/uproxy/uproxy/2313#-KDTxQSnnudhb4ig8wc8:-KDTxQSnnudhb4ig8wc9:-998253164) ([raw file](https://github.com/uproxy/uproxy/blob/014cf92c9c9fec420808e0f5aa975ff1720dba60/src/generic_ui/locales/en/messages.json#L1192)):*
Capitalise "we".

---

*[src/generic_ui/locales/en/messages.json, line 1212 \[r1\]](https://reviewable.io:443/reviews/uproxy/uproxy/2313#-KDTy8qfadZFqusm4jvK:-KDTy8qfadZFqusm4jvL:1806819585) ([raw file](https://github.com/uproxy/uproxy/blob/014cf92c9c9fec420808e0f5aa975ff1720dba60/src/generic_ui/locales/en/messages.json#L1212)):*
Perhaps more of a UX question but it seems to me like "connect" should match the translation for`SIGN_IN_AND_CONNECT`.

If so, I think it will help the translators to mention this in the description.

---

*[src/generic_ui/locales/en/messages.json, line 1220 \[r1\]](https://reviewable.io:443/reviews/uproxy/uproxy/2313#-KDTzAiQaQtaZK2krorf:-KDTzAiQaQtaZK2krorg:-247486815) ([raw file](https://github.com/uproxy/uproxy/blob/014cf92c9c9fec420808e0f5aa975ff1720dba60/src/generic_ui/locales/en/messages.json#L1220)):*
Again, this might be a UX question but here we're again referencing the name of a button - doubly hard in this case because it's not out service and we can't easily check the translation in different languages.

---

*[src/generic_ui/locales/en/messages.json, line 1224 \[r1\]](https://reviewable.io:443/reviews/uproxy/uproxy/2313#-KDTzmrTT0gipzkzUFzX:-KDTzmrTT0gipzkzUFzY:-2146763930) ([raw file](https://github.com/uproxy/uproxy/blob/014cf92c9c9fec420808e0f5aa975ff1720dba60/src/generic_ui/locales/en/messages.json#L1224)):*
The mocks for this screen include a full stop.

---

*[src/generic_ui/locales/en/messages.json, line 1248 \[r1\]](https://reviewable.io:443/reviews/uproxy/uproxy/2313#-KDTu_lA1AZIIlC4bp8r:-KDTu_lA1AZIIlC4bp8s:-189831155) ([raw file](https://github.com/uproxy/uproxy/blob/014cf92c9c9fec420808e0f5aa975ff1720dba60/src/generic_ui/locales/en/messages.json#L1248)):*
Again, may be a UX question but is there some reason for ampersands here? It doesn't seem to match the style of the rest of the strings.

---

*[src/generic_ui/polymer/button.html, line 7 \[r1\]](https://reviewable.io:443/reviews/uproxy/uproxy/2313#-KDU48d6qXr5zfqz_x_U:-KDU48d6qXr5zfqz_x_V:-128473454) ([raw file](https://github.com/uproxy/uproxy/blob/014cf92c9c9fec420808e0f5aa975ff1720dba60/src/generic_ui/polymer/button.html#L7)):*
is it still teal 500?

---

*[src/generic_ui/polymer/cloud-install.html, line 92 \[r1\]](https://reviewable.io:443/reviews/uproxy/uproxy/2313#-KDU5LICT2bNLBD4tQZ7:-KDU5LICT2bNLBD4tQZ8:977858439) ([raw file](https://github.com/uproxy/uproxy/blob/014cf92c9c9fec420808e0f5aa975ff1720dba60/src/generic_ui/polymer/cloud-install.html#L92)):*
This isn't a large block, I think you could get away without the `end of .content` comment (generally a smell).

---

*[src/generic_ui/polymer/cloud-install.html, line 114 \[r1\]](https://reviewable.io:443/reviews/uproxy/uproxy/2313#-KDU5SCVaS4pi0REy9fn:-KDU5SCVaS4pi0REy9fo:-2090249022) ([raw file](https://github.com/uproxy/uproxy/blob/014cf92c9c9fec420808e0f5aa975ff1720dba60/src/generic_ui/polymer/cloud-install.html#L114)):*
I think you want `{{ 'SIGN_ME_IN' | $$}`.

---

*[src/generic_ui/polymer/cloud-install.html, line 132 \[r1\]](https://reviewable.io:443/reviews/uproxy/uproxy/2313#-KDU5eQcK3AAqFwwDPaE:-KDU5eQcK3AAqFwwDPaF:-2068738740) ([raw file](https://github.com/uproxy/uproxy/blob/014cf92c9c9fec420808e0f5aa975ff1720dba60/src/generic_ui/polymer/cloud-install.html#L132)):*
How about translating the regions?

---

*[src/generic_ui/polymer/cloud-install.ts, line 13 \[r1\]](https://reviewable.io:443/reviews/uproxy/uproxy/2313#-KDUFki2Paxe_aS3T5Hu:-KDUFki3yIjwOTl3uuYo:-1055559291) ([raw file](https://github.com/uproxy/uproxy/blob/014cf92c9c9fec420808e0f5aa975ff1720dba60/src/generic_ui/polymer/cloud-install.ts#L13)):*
Suggestion: `open`

---

*[src/generic_ui/polymer/cloud-install.ts, line 28 \[r1\]](https://reviewable.io:443/reviews/uproxy/uproxy/2313#-KDUG-AMkrEgmNwNgjuH:-KDUG-AMkrEgmNwNgjuI:-2047727088) ([raw file](https://github.com/uproxy/uproxy/blob/014cf92c9c9fec420808e0f5aa975ff1720dba60/src/generic_ui/polymer/cloud-install.ts#L28)):*
I think you can simplify this to:
```typescript
back: function() {
  if (this.$.digitalOceanAccountHelpOverlay.opened ||
      this.$.loginOverlay.opened || this.$.failureOverlay.opened) {
    this.closeOverlays();
    this.$.getStartedOverlay.open();
  } else {
    this.closeOverlays();
  }
}
```

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/uproxy/uproxy/2313#-:-KDTtO_1PVEOKrUWYg2G:465328456)*
<!-- Sent from Reviewable.io -->

dborkan commented:
Thanks, just made those changes plus I changed a number of strings to match the latest mocks

---

Review status: 0 of 35 files reviewed at latest revision, 19 unresolved discussions.

---

*[src/generic_ui/locales/en/messages.json, line 1188 \[r1\]](https://reviewable.io:443/reviews/uproxy/uproxy/2313#-KDTynB9lJu3IiuLVm4l:-KDV9iQ6lepQP5ugGh0q:-1139496375) ([raw file](https://github.com/uproxy/uproxy/blob/014cf92c9c9fec420808e0f5aa975ff1720dba60/src/generic_ui/locales/en/messages.json#L1188)):*
Yeah that is standard throughout this file, although HTML often removes one of these spaces.  I think it's fine for now

---

*[src/generic_ui/locales/en/messages.json, line 1220 \[r1\]](https://reviewable.io:443/reviews/uproxy/uproxy/2313#-KDTzAiQaQtaZK2krorf:-KDVAN2PXaZuL1zzK0jQ:1857389017) ([raw file](https://github.com/uproxy/uproxy/blob/014cf92c9c9fec420808e0f5aa975ff1720dba60/src/generic_ui/locales/en/messages.json#L1220)):*
Just updated the description

---

*[src/generic_ui/locales/en/messages.json, line 1248 \[r1\]](https://reviewable.io:443/reviews/uproxy/uproxy/2313#-KDTu_lA1AZIIlC4bp8r:-KDVAScHB5RGRqpKGd0I:1755104638) ([raw file](https://github.com/uproxy/uproxy/blob/014cf92c9c9fec420808e0f5aa975ff1720dba60/src/generic_ui/locales/en/messages.json#L1248)):*
This is how it was in the mocks I was given, I think maybe it's to save space?

---

*[src/generic_ui/polymer/button.html, line 7 \[r1\]](https://reviewable.io:443/reviews/uproxy/uproxy/2313#-KDU48d6qXr5zfqz_x_U:-KDVAZWh9yDMLsoDGyvL:-1558130035) ([raw file](https://github.com/uproxy/uproxy/blob/014cf92c9c9fec420808e0f5aa975ff1720dba60/src/generic_ui/polymer/button.html#L7)):*
Just removed this

---

*[src/generic_ui/polymer/cloud-install.html, line 132 \[r1\]](https://reviewable.io:443/reviews/uproxy/uproxy/2313#-KDU5eQcK3AAqFwwDPaE:-KDVEvrzMR2D-4MNiMeQ:937413295) ([raw file](https://github.com/uproxy/uproxy/blob/014cf92c9c9fec420808e0f5aa975ff1720dba60/src/generic_ui/polymer/cloud-install.html#L132)):*
Done.  I also removed displaying the region on the success screen since that's no longer in the mocks and it was a pain to map from the DigitalOcean code to the label (e.g. nyc2 to "New York")

---

*[src/generic_ui/polymer/cloud-install.ts, line 28 \[r1\]](https://reviewable.io:443/reviews/uproxy/uproxy/2313#-KDUG-AMkrEgmNwNgjuH:-KDVB-0P1GLvSFDh__Ww:657014608) ([raw file](https://github.com/uproxy/uproxy/blob/014cf92c9c9fec420808e0f5aa975ff1720dba60/src/generic_ui/polymer/cloud-install.ts#L28)):*
Just simplified this a bit more

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/uproxy/uproxy/2313#-:-KDVGBSJJ-ELN9ARgwN4:-1374505040)*
<!-- Sent from Reviewable.io -->

trevj commented:
Review status: 0 of 35 files reviewed at latest revision, 8 unresolved discussions.

---

*[src/generic_ui/locales/en/messages.json, line 1188 \[r1\]](https://reviewable.io:443/reviews/uproxy/uproxy/2313#-KDTynB9lJu3IiuLVm4l:-KDYuXcFfTQMil3nKlNR:1287971648) ([raw file](https://github.com/uproxy/uproxy/blob/014cf92c9c9fec420808e0f5aa975ff1720dba60/src/generic_ui/locales/en/messages.json#L1188)):*
Weird. Okay.

---

*[src/generic_ui/polymer/cloud-install.ts, line 28 \[r1\]](https://reviewable.io:443/reviews/uproxy/uproxy/2313#-KDUG-AMkrEgmNwNgjuH:-KDYuPYlQEAqIMTYllej:-865400185) ([raw file](https://github.com/uproxy/uproxy/blob/014cf92c9c9fec420808e0f5aa975ff1720dba60/src/generic_ui/polymer/cloud-install.ts#L28)):*
If you close the overlays first, will the `.opened` check still work?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/uproxy/uproxy/2313)*
<!-- Sent from Reviewable.io -->

dborkan commented:
Review status: 0 of 35 files reviewed at latest revision, 7 unresolved discussions.

---

*[src/generic_ui/polymer/cloud-install.ts, line 28 \[r1\]](https://reviewable.io:443/reviews/uproxy/uproxy/2313#-KDUG-AMkrEgmNwNgjuH:-KDZPaH13FlVc7M7Eczl:-267606704) ([raw file](https://github.com/uproxy/uproxy/blob/014cf92c9c9fec420808e0f5aa975ff1720dba60/src/generic_ui/polymer/cloud-install.ts#L28)):*
Ah you are right, just changed it to your suggestion

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/uproxy/uproxy/2313)*
<!-- Sent from Reviewable.io -->

trevj commented:
Yay!

:+1:

---

Review status: 0 of 35 files reviewed at latest revision, 7 unresolved discussions, some commit checks broke.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/uproxy/uproxy/2313#-:-KDZoLVbj5XOCYHklV4U:-1016335089)*
<!-- Sent from Reviewable.io -->
